h2/src/main/org/h2/mvstore/db/MVSecondaryIndex.java

```diff
@@ -252,6 +252,13 @@ public void remove(Session session, Row row) {
         }
     }
 
+    @Override
+    public void update(Session session, Row oldRow, Row newRow) {
+        if (compareRows(oldRow, newRow ) != 0) {
```

Comment:
Perhaps it doesn't mean that rows are really identical to each other, but only that they have the same place in sorting order or I'm missing something? If this method will be used wider someday this may lead to usage of old values with `NUMERIC`, `VARCHAR_IGNORECASE`, `TIMESTAMP WITH TIME ZONE`, or may be some other data type that has distinct values with `compareTo(…) == 0`.